core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -479,6 +491,23 @@ public ViewMetadata build() {
           metadataLocation);
     }
 
+    private void removeIntermediateViewVersionIdChanges() {
```

Comment:
It looks like I was wrong about the way `TableMetadata` works. I don't think it is rewriting the set of changes in the build method after all. I think we should do that as well to keep it simple and avoid a few problems with doing it this way (`removeIf` will not remove duplicates if they set the same version as current).